tensorflow/stream_executor/rocm/rocm_driver.cc

```diff
@@ -1254,9 +1256,26 @@ static port::StatusOr<T> GetSimpleAttribute(hipDevice_t device,
     LOG(ERROR) << "failed to query device memory info: " << ToString(res);
     return false;
   }
-
-  *free_out = free;
-  *total_out = total;
+  // On gfx90a, we hide 1 GB of GPU memory from TF, to allow for late
+  // allocations by internal ROCm libraries (e.g. rocBLAS alone needs
+  // ~200 MB to put its kernels as of ROCm 4.1)
+  uint64 reserve = 0;
+  hipDeviceProp_t props;
+  hipDevice_t dev;
+  res = hipGetDevice(&dev);
+  if(res == hipSuccess) {
+    res = tensorflow::wrap::hipGetDeviceProperties(&props, dev);
+    if (res == hipSuccess) {
+      std::string gcnArchName = props.gcnArchName;
+      if (gcnArchName.substr(0,6)=="gfx90a" || gcnArchName.substr(0,6)=="gfx910")
+        reserve = 1048576*1024;
```

Comment:
perhaps separate this out into a separate func that takes the gfx arch as input, and returns the amount of memory to reserve / hide from TF as output

we can handle this in code too - https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/ci_build/Dockerfile.rocm#L114

Reply:
I'll separate this out. I don't want to rely on being inside an "official" TF docker container.